src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -73,13 +74,13 @@ protected function displaySingleMessage(Envelope $envelope, SymfonyStyle $io)
             $rows[] = ['Message Id', $id];
         }
 
-        $flattenException = null === $lastRedeliveryStamp ? null : $lastRedeliveryStamp->getFlattenException();
+        $flattenException = null === $lastRedeliveryStampWithException ? null : $lastRedeliveryStampWithException->getFlattenException();
         if (null === $sentToFailureTransportStamp) {
             $io->warning('Message does not appear to have been sent to this transport after failing');
         } else {
             $rows = array_merge($rows, [
                 ['Failed at', null === $lastRedeliveryStamp ? '' : $lastRedeliveryStamp->getRedeliveredAt()->format('Y-m-d H:i:s')],
```

Comment:
Leaving this as `$lastRedeliveryStamp` is a little dishonest: we're showing "Failed on Dec 1st", but then the exception we show might be from Nov 30th. Probably we should use `$lastRedeliveryStampWithException` everywhere to at least show consistent information. The full retry history is already displayed below under "Message History", so that should help.

Reply:
Good point. I updated it.